Re-organize install guide and platforms page (#1637)
sensu/sensu-docs: content/sensu-go/5.11/installation/install-sensu.md

weight: 1
version: "5.11"
product: "Sensu Go"
menu:
  sensu-go-5.11:
    parent: installation
---

Sensu Go is available for Linux, Windows (agent and CLI only), macOS (CLI only), and Docker.
If you’re trying out Sensu for the first time, we recommend setting up a local environment using the [Sensu sandbox][14].
If you’re deploying Sensu to your infrastructure, we recommend one of our supported packages, Docker images, or [configuration management integrations][15].
Sensu downloads are provided under the [Sensu License][13]; see the [supported platforms page][5] for more information.

- [Install sensuctl](#install-sensuctl)
- [Install the Sensu backend](#install-the-sensu-backend)
- [Install Sensu agents](#install-sensu-agents)
- [Activate licensed-tier features](#activate-licensed-tier-features)

### Architecture overview

<img src="/images/install-sensu.svg" alt="Sensu architecture diagram">

Powered by an an embedded transport and [etcd][16] datastore, the **Sensu backend** gives you flexible, automated workflows to route metrics and alerts.

> Sensu backends require persistent storage, a local cache, and three exposed ports:

Reviewer: ```suggestion
Sensu backends require persistent storage for their embedded database, disk space for local asset caching, and three exposed ports:
```

- `3000` - Sensu [web UI][25]

> - `8080` - Sensu [API][26] (and, by extension, connections via sensuctl)

Reviewer: ```suggestion
- `8080` - Sensu [API][26] used by sensuctl, some plugins, and any of your custom tooling
```

- `8081` - WebSocket API used by Sensu agents

Sensu backends running in a [clustered configuration][22] require additional ports.
See the [hardware requirements][hardware] guide for deployment recommendations.
<!-- Link to deployment guide -->

**Sensu agents** are lightweight clients that run on the infrastructure components you want to monitor.
Agents register automatically with Sensu as entities and are responsible for creating check and metric events to send to the backend event pipeline.
Optionally, agents can expose ports `3031` for the [agent API][27] and `8125` for the [StatsD listener][28].

> Agents using Sensu [assets][17] require a local cache.

Reviewer: ```suggestion
Agents using Sensu [assets][17] require some disk space for a local cache.
```

### Install sensuctl

> Sensuctl is a command line tool for managing resources within Sensu. It works by calling Sensu’s underlying API to create, read, update, and delete resources, events, and entities. Sensuctl is available for Linux, Windows, and macOS.

Reviewer: ```suggestion
Sensuctl is a command line tool for managing resources within Sensu. It works by calling Sensu’s HTTP API to create, read, update, and delete resources, events, and entities. Sensuctl is available for Linux, Windows, and macOS.
```

See the [supported platforms page][5] for more information.

##### 1. Download

{{< language-toggle >}}

{{< highlight Docker >}}
# All Sensu Docker images contain a Sensu backend and a Sensu agent

# Pull the Alpine-based image
docker pull sensu/sensu

# Pull the image based on Red Hat Enterprise Linux
docker pull sensu/sensu-rhel
{{< /highlight >}}

{{< highlight "Ubuntu/Debian" >}}
# Add the Sensu repository
curl -s https://packagecloud.io/install/repositories/sensu/stable/script.deb.sh | sudo bash

# Install the sensu-go-backend package
sudo apt-get install sensu-go-backend
{{< /highlight >}}

{{< highlight "RHEL/CentOS" >}}
# Add the Sensu repository
curl -s https://packagecloud.io/install/repositories/sensu/stable/script.rpm.sh | sudo bash

# Install the sensu-go-backend package
sudo yum install sensu-go-backend
{{< /highlight >}}

{{< /language-toggle >}}

##### 2. Configure and start

You can configure Sensu using `sensu-backend start` flags or an /etc/sensu/backend.yml file, the former taking precedence.
At a minimum, the Sensu backend requires the `state-dir` flag, but here are some other useful configs and templates.

{{< language-toggle >}}

{{< highlight Docker >}}

> docker run -v /var/lib/sensu:/var/lib/sensu -d --name sensu-backend -p 2380:2380 -p 3000:3000 -p 8080:8080 -p 8081:8081 sensu/sensu:latest sensu-backend start

Reviewer 2: You can skip the volume mount here. It's really only if you're planning on needing to have cached assets & such. Useful for demos, less so in prod, I'd imagine. Otherwise, LGTM

Author: @cwjohnston recommended that we keep the volume mount here, but we can always revisit.

{{< /highlight >}}

{{< highlight "Docker Compose" >}}
---
version: "3"
services:
  sensu-backend:
    image: sensu/sensu:latest
    ports:
    - 3000:3000
    - 8080:8080
    - 8081:8081
    volumes:
    - "sensu-backend-data:/var/lib/sensu/etcd"
    command: "sensu-backend start --state-dir /var/lib/sensu/sensu-backend --log-level debug"

volumes:
  sensu-backend-data:
    driver: local

> {{< /highlight >}}

Author: To do:

- [ ] Make sure `docker run` example is correct and reflects best practices
- [ ] Make sure Docker Compose example is consistent with `docker run` example

# Start the installation wizard
msiexec.exe /i $env:userprofile\sensu-go-agent_5.11.0.4171_en-US.x64.msi
{{< /highlight >}}

{{< /language-toggle >}}

##### 2. Configure and start {#agent-start}

You can configure the Sensu agent using `sensu-agent start` flags or an /etc/sensu/agent.yml file, the former taking precedence.
At a minimum, the Sensu agent requires the `--backend-url` flag, but here are some other useful configs and templates.

{{< language-toggle >}}

{{< highlight Docker >}}
# If you are running the agent locally on the same system as the Sensu backend, add `--link sensu-backend` to your Docker arguments and change the backend URL to `--backend-url ws://sensu-backend:8081`.
docker run -v /var/lib/sensu:/var/lib/sensu -d --name sensu-agent sensu/sensu:latest sensu-agent start --backend-url ws://sensu.yourdomain.com:8081 --subscriptions webserver,system --cache-dir /var/lib/sensu
{{< /highlight >}}

{{< highlight "Docker Compose" >}}
---
version: "3"
services:
  sensu-agent:
    image: sensu/sensu:latest
    ports:
    - 3031:3031
    command: "sensu-agent start --backend-url ws://sensu-backend:8081 --log-level debug --subscriptions webserver --api-host 0.0.0.0"

> {{< /highlight >}}

Author: To do:

- [x] Make sure `docker run` example is correct and reflects best practices
- [x] Make sure Docker Compose example is consistent with `docker run` example

Reviewer 2: LGTM 👍

sudo service sensu-backend start

# Verify that the backend is running
sudo service sensu-backend status
{{< /highlight >}}

{{< highlight "RHEL/CentOS" >}}
# Copy the config template from the docs
sudo curl -L https://docs.sensu.io/sensu-go/latest/files/backend.yml -o /etc/sensu/backend.yml

# Start sensu-backend using a service manager
sudo service sensu-backend start

# Verify that the backend is running
sudo service sensu-backend status
{{< /highlight >}}

{{< /language-toggle >}}

For a complete list of config options, see the [backend reference][6].

### 3. Open the Sensu web UI

The web UI provides a unified view of your monitoring events with user-friendly tools to reduce alert fatigue.
After starting the Sensu backend, open the web UI by visiting http://localhost:3000.
You may need to replace `localhost` with the
hostname or IP address where the Sensu backend is running.

To log in, enter your Sensu user credentials, or use Sensu's default admin credentials: username: `admin` and password: `P@ssw0rd!`.
For now, you should see a blank slate: no incidents or entities.
Select the ☰ icon to explore the web UI.

### 4. Make a request to the Sensu health API

> To make sure the backend is up and running, we'll create check the health of the backend using the Sensu API.

Reviewer 3: ```suggestion
To make sure the backend is up and running, we'll check the health of the backend using the Sensu API.
```

{{< highlight shell >}}
curl http://127.0.0.1:8080/health
{{< /highlight >}}

> You should see a 200 OK response from the API and `"Healthy": true` if the backend is up and running.

Reviewer 3: These might both be nitpicks, but you don't actually see the return code with the current `curl` command. Do we want to add the flag that shows the header? I suppose there is an assumption there if it does return that it is a 200 OK. 

Also, more than just `"Healthy": true` is outputted. Maybe say the output will include this message if the idea here is to keep the install guide as minimal as possible vs showing the entire output, which will differ anyways (cluster_id, member_id).

Author: Fixed in https://github.com/sensu/sensu-docs/pull/1637/commits/06ae7852ea0ff15d352207efccaa9a2e5bd4d38c

curl https://packagecloud.io/install/repositories/sensu/stable/script.deb.sh | sudo bash

# Install the sensu-go-backend package
sudo apt-get install sensu-go-agent
{{< /highlight >}}

{{< highlight "RHEL/CentOS" >}}
# Add the Sensu repository
curl https://packagecloud.io/install/repositories/sensu/stable/script.rpm.sh | sudo bash

# Install the sensu-go-backend package
sudo yum install sensu-go-agent
{{< /highlight >}}

{{< highlight "Windows" >}}
# Download the Sensu agent for Windows amd64
Invoke-WebRequest https://s3-us-west-2.amazonaws.com/sensu.io/sensu-go/5.11.0/sensu-go-agent_5.11.0.4171_en-US.x64.msi  -OutFile "$env:userprofile\sensu-go-agent_5.11.0.4171_en-US.x64.msi"
# Or for 386
Invoke-WebRequest https://s3-us-west-2.amazonaws.com/sensu.io/sensu-go/5.11.0/sensu-go-agent_5.11.0_2380_en-US.x86.msi  -OutFile "$env:userprofile\sensu-go-agent_5.11.0.4171_en-US.x86.msi"

# Start the installation wizard
msiexec.exe /i $env:userprofile\sensu-go-agent_5.11.0.4171_en-US.x64.msi
{{< /highlight >}}

{{< /language-toggle >}}

### 2. Configure and start

You can configure the Sensu agent using `sensu-agent start` flags or an /etc/sensu/agent.yml file, the former taking precedence.

> At a minimum, the Sensu agent requires the `-backend-url` flag, but here are some other useful configs and templates.

Reviewer 3: ```suggestion
At a minimum, the Sensu agent requires the `--backend-url` flag, but here are some other useful configs and templates.
```

Or remove the single `-`, not sure which is more correct here.
